app/lib/service/security_advisories/backend.dart

```diff
@@ -212,6 +212,28 @@ List<String> sanityCheckOSV(OSV osv) {
     errors.add('Invalid id, id too long (over 255 characters).');
   }
 
+  if (osv.affected == null || osv.affected!.isEmpty) {
+    errors.add('No affected packages for advisory ${osv.id}');
+  } else {
+    bool noAffectedHasVersion = true;
+    osv.affected!
+        .where((a) => a.package.ecosystem.toLowerCase() == 'pub')
+        .forEach((affected) {
+      if (affected.versions == null || affected.versions!.isEmpty) {
+        logger.warning('No versions specified for affected package '
+            '${affected.package.name} in advisory ${osv.id}.');
+      } else {
+        noAffectedHasVersion = false;
+      }
+    });
```

Comment:
Just an idea, I haven't tested it. But it will find package names that are mentioned as being affected, but for which there isn't any of the entries in `affected` that specifies an explicit list of versions.
```suggestion
    final affectedPackagesWithoutVersions = osv.affected!
        .where((a) => a.package.ecosystem.toLowerCase() == 'pub')
        .groupBy((a) => a.package.name)
        .entries
        .where(((:key, :value)) => value.every((a) => (a.versions ?? []).isEmpty))
        .map(((:key, :value)) => key)
        .toSet();
    if (affectedPackagesWithoutVersions.isNotEmpty) {
      logger.warning(
        'Advisory ${osv.id} affects packages: '
        '${affectedPackagesWithoutVersions.join(', ')}, '
        'but does not specify explicit versions.',
      );
    }
```

Reply:
I think it is preferrable to have a warning in the log for each object that does not have versions. In practice I doubt, this will happen very often though.

Also I think it is important with the error if non of the affected packages has `versions`, since then the advisory is not relevant.